diversity_analysis_tool/graph_construction.py

```diff
@@ -17,39 +18,117 @@ def __init__(self, df, output_directory_path):
-            self.generate_stacked_bar_graph('age_band', 'ethnicity', 'Number of participants', colname_dict['age_band'],
-                                            colname_dict['ethnicity'])
-        if set(['age_band', 'race']).issubset(self.df.columns.values):
-            self.generate_stacked_bar_graph('age_band', 'race', 'Number of participants', colname_dict['age_band'],
-                                            colname_dict['race'])
-        if set(['race', 'sex']).issubset(self.df.columns.values):
-            self.generate_stacked_bar_graph('race', 'sex', 'Number of participants', colname_dict['race'],
-                                            colname_dict['sex'])
-        if set(['ethnicity', 'sex']).issubset(self.df.columns.values):
-            self.generate_stacked_bar_graph('ethnicity', 'sex', 'Number of participants', colname_dict['ethnicity'],
-                                            colname_dict['sex'])
-        if set(['age_band', 'sex']).issubset(self.df.columns.values):
-            self.generate_stacked_bar_graph('age_band', 'sex', 'Number of participants', colname_dict['age_band'],
-                                            colname_dict['sex'])
+        if set(["age_band", "ethnicity"]).issubset(self.df.columns.values):
+            self.generate_stacked_bar_graph(
+                "age_band",
+                "ethnicity",
+                "Number of participants",
+                colname_dict["age_band"],
+                colname_dict["ethnicity"],
+            )
+        if set(["age_band", "race"]).issubset(self.df.columns.values):
+            self.generate_stacked_bar_graph(
+                "age_band",
+                "race",
+                "Number of participants",
+                colname_dict["age_band"],
+                colname_dict["race"],
+            )
+        if set(["race", "sex"]).issubset(self.df.columns.values):
+            self.generate_stacked_bar_graph(
+                "race",
+                "sex",
+                "Number of participants",
+                colname_dict["race"],
+                colname_dict["sex"],
+            )
+        if set(["ethnicity", "sex"]).issubset(self.df.columns.values):
+            self.generate_stacked_bar_graph(
+                "ethnicity",
+                "sex",
+                "Number of participants",
+                colname_dict["ethnicity"],
+                colname_dict["sex"],
+            )
+        if set(["age_band", "sex"]).issubset(self.df.columns.values):
+            self.generate_stacked_bar_graph(
+                "age_band",
+                "sex",
+                "Number of participants",
+                colname_dict["age_band"],
+                colname_dict["sex"],
+            )
+
+        self.plot_missing_rates(
+            colname_dict, x_label="% of entries missing", show_percentage=True
+        )
+
+    def plot_missing_rates(
+        self, colname_dict, x_label=None, y_label=None, show_percentage=True
+    ):
```

Comment:
Should we call it `show_fraction` instead of percentage?

Reply:
yes good idea!